cmd/rebootleby/src/lib.rs

```diff
@@ -0,0 +1,274 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility rebootleby`
+//!
+//! You don't want to use this right now.
+
+use anyhow::{anyhow, bail, Context, Result};
+use clap::{CommandFactory, Parser};
+use humility::core::Core;
+use humility_cli::{ExecutionContext, Subcommand};
+use humility_cmd::{Archive, Attach, Command, CommandKind, Validate};
+use std::io::Read;
+use std::ops::RangeInclusive;
+use std::path::PathBuf;
+use std::time::Duration;
+use zip::ZipArchive;
+
+#[derive(Parser, Debug)]
+#[clap(about = env!("CARGO_PKG_DESCRIPTION"))]
+struct Rebootleby {
+    /// Path to a Bootleby Bundle file.
+    path: PathBuf,
+}
+
+fn rebootleby(context: &mut ExecutionContext) -> Result<()> {
+    let Subcommand::Other(subargs) = context.cli.cmd.as_ref().unwrap();
+    let subargs = Rebootleby::try_parse_from(subargs)?;
+
+    // Load bundle
+    let bundle_reader = std::fs::File::open(&subargs.path)
+        .with_context(|| format!("loading {}", subargs.path.display()))?;
+    let mut zip =
+        ZipArchive::new(bundle_reader).context("opening bundle file as ZIP")?;
+
+    let img_bootleby = {
+        let mut entry = zip
+            .by_name("bootleby.bin")
+            .context("can't find bootleby.bin in bundle")?;
+        let mut data = vec![];
+        entry.read_to_end(&mut data).context("reading bootleby.bin")?;
+        data
+    };
+    let img_cmpa = {
+        let mut entry =
+            zip.by_name("cmpa.bin").context("can't find cmpa.bin in bundle")?;
+        let mut data = vec![];
+        entry.read_to_end(&mut data).context("reading cmpa.bin")?;
+        data
+    };
+    let img_cfpa = {
+        let mut entry =
+            zip.by_name("cfpa.bin").context("can't find cfpa.bin in bundle")?;
+        let mut data = vec![];
+        entry.read_to_end(&mut data).context("reading cfpa.bin")?;
+        data
+    };
+
+    let img_cmpa: &[u8; 512] = img_cmpa[..]
+        .try_into()
+        .map_err(|_| anyhow!("CMPA file is wrong length!"))?;
+    let img_cfpa: &[u8; 512] = img_cfpa[..]
+        .try_into()
+        .map_err(|_| anyhow!("CFPA file is wrong length!"))?;
+
+    println!("Loaded bundle {}", subargs.path.display());
```

Comment:
Nit: using `humility::msg!` would be more idiomatic (same for all `println!` in this file)